_posts/2017-07-05-SpringIO17-Summary.md

```diff
@@ -0,0 +1,773 @@
+- Seamless **Spring Security Integration**
+- Supports OAuth 2.0, SAML 2.0, OpenID Connect
+- Integration with Active Directory, LDAP and even Kerberos (*start drooling enterprise users!*)
+
+It's actually quite easy to setup Keycloak:
+
+- Download the Keycloak standalone server [here](https://downloads.jboss.org/keycloak/3.2.0.Final/keycloak-3.2.0.Final.zip){:target="_blank"}
+- Extract and run it
+- Start the server and create an admin user
+- Create a new realm
+- Create a new application
+- Add roles to your application
+- Create a user to authenticate with
+- Create a Spring Boot application at [https://start.spring.io](start.spring.io){:target="_blank"}) and include the Keycloak starter
+- Add the Keycloak properties to your application.yml:
+    - server URL
+    - realm
+    - resource name of your application
+    - security constraints for your users
+- Run!
+
+There are many additional features for power users:
+
+- Automatic registration of applications should be possible using a one-time token (coming soon?)
+- Centralized User Management
+- **CORS support** for Single Page Applications
+- Social Login Integration
+- Registration and Forgot Password functionality, all out-of-the-box, configurable at runtime
+- UI Customization of all pages is possible through theming (*start drooling designers!*)
+
+All in all, the setup and demo went very smooth and **I genuinely feel this product is about to become very popular**, partly because of the Spring Boot integration, but also because it just seems very solid and user-friendly.
+There might be a dedicated blogpost coming soon about Keycloak, so stay tuned and check our blog regularly or subscribe to our [RSS feed](https://ordina-jworks.github.io/feed.xml){:target="_blank"}!
+
+# Spring Cloud Streams (Data Services)
+
+### by [Michael Minella](https://twitter.com/michaelminella){:target="_blank"}
+
+Michael gave a summary about all the new projects in the Spring ecosystem that process data and / or message very well.
+He explained that there are lots of big data frameworks out there (Hadoop, Spark, ...), which can handle BIG amounts of data very well.
+However, they are usually **too bulky / difficult / inappropriate for handling smaller volumes of data**.
+Also, for quickly setting up something like Hadoop or Spark, the **learning curve is too high** and the effort doesn't justify the benefits.
+
+#### Solution: data microservices
+
+  - Developed and tested in isolation, also easier to test
+  - Independently scalable depending on data processing load
+  - Familiar development model, just like regular cloud-native Spring microservices
+  - Easier to govern for Ops
+  - So the need for data / app integration / composition arises
+  - Which means the need for orchestration and operational coverage arises (lots of plumbing required)
+
+#### Spring Cloud Stream
```

Comment:
add link to the Spring Cloud project pages?
like: https://cloud.spring.io/spring-cloud-stream/